doc/data-warehouse-schema.md

```diff
@@ -0,0 +1,311 @@
+                - [Benefits](#benefits)
+                - [Drawbacks](#drawbacks)
+
+<!-- markdown-toc end -->
+
+# Data warehouse schema
+
+This document describes the database schema for the data warehouse. This is
+technical documentation so it is recommended to read first [what is a data
+wareouse and why GOV.UK uses one][data-warehouse-what-why] in order to get a
+better understanding of the problem that the data warehouse aims to solve.
+
+## ERD - schema ##
+
+The image below represents the schema for the data warehouse. There are not many
+tables but some of them are quite big. It is imporant to understand their
+associations and the type of data that each table stores.
+
+![data warehouse Database Schema][database-schema-image]
+
+### How to generate the schema ###
+
+The schema is generated using [Rails ERD][rails-erd] running the following
+command:
+
+```bash
+$ bundle exec erd --filetype=png --connected --title 'Schema' --filename doc/images/schema
+```
+
+*Developer note*: The schema might not be an exact representation of the latest
+version of the schema, but it should be close enough to get an understanding of
+it.
+
+## Introduction
+
+Data warehouses use a different design from standard operational databases. The
+latter are optimized to maintain strict accuracy of data by updating real-time
+information. Data warehouses, by contrast, are designed to give a long-range
+view of data over time. They trade off transaction volume and instead specialize
+in data aggregation.
+
+The data warehouse for GOV.UK stores data collected from other systems
+([Feedex][feedex], [Google Analytics][google analytics], ...) which are mostly
+backed by operational databases and stores this information using a `star
+schema` database design.
+
+```
+The star schema consists of one or more fact tables referencing any number of
+dimension tables; it separates data into `facts` and `dimensions`:
+```
+
+### Facts tables ###
+
+A fact table is the central table in a star schema of a data warehouse. A fact
+table stores quantitative information for analysis and is often denormalized.
+
+```
+Record meassurements or metrics for specific events. They usually consist of
+numeric values and foreign keys to dimensional data.
+```
+
+A fact table consists of two types of columns (foreign keys and meassures).
+
+The **foreign keys** column allows joins with dimension tables, and the
+**measures** columns contain the data that is being analyzed.
+
+In GOV.UK the facts tables stores metrics about content. These metrics are
+collected on a daily basis or with each content change.
+
+### Dimensional tables ###
+
+Define facts characteristics, and usually have a relatively small number of
+records, compared with the facts tables, but higher number of attributes
+(columns) for filtering and grouping.
+
+```
+Categorizes facts and measures to enable users to answer business questions.
+```
```

Comment:
Example columns for dimensions and facts tables may help these sections.

Reply:
There are some samples below, when I talk in more detail about each table.